src/main/java/com/connexta/store/adaptors/StorageAdaptor.java

```diff
@@ -39,4 +39,12 @@ void store(
    */
   @NotNull
   StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key) throws RetrieveException;
+
+  /**
+   * Updates the {@link StoreStatus} of {@code dataset} for a given {@code datasetId}
+   *
+   * @param datasetId the Id of a dataset
+   * @param storeStatus the new {@link StoreStatus} for a dataset
+   */
+  void updateStatus(String datasetId, String storeStatus);
```

Comment:
❓ Does this have to be called after `store()`?

Reply:
No, store adds the tag in its request

Reply:
Ah I see now.